src/Plugin/Block/CorporateFormBlock.php

```diff
@@ -0,0 +1,147 @@
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFormBuilderInterface $entity_form_builder
+   *   The entity form builder.
+   */
+  public function __construct(array $configuration, string $plugin_id, array $plugin_definition, EntityTypeManagerInterface $entityTypeManager, EntityFormBuilderInterface $entity_form_builder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityFormBuilder = $entity_form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity.form_builder')
+    );
+  }
+
+  /**
+   * When the contact form is modified, these cache tags will be invalidated.
+   */
+  public function getCacheTags() {
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $this->getContactForm();
+    if (!$contact_form) {
+      return [];
+    }
+
+    return Cache::mergeTags(parent::getCacheTags(), $contact_form->getCacheTags());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockAccess(AccountInterface $account) {
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $this->getContactForm();
+
+    if (!$contact_form) {
+      return AccessResult::forbidden();
+    }
+
+    $access = $this->entityTypeManager->getAccessControlHandler('contact_message')->createAccess($contact_form->id(), $account);
+
+    if (!$access) {
+      return AccessResult::forbidden()->cachePerPermissions();
+    }
+
+    return AccessResult::allowed()->cachePerPermissions()->addCacheTags($contact_form->getCacheTags());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    /** @var \Drupal\contact\ContactFormInterface $contact_form */
+    $contact_form = $this->getContactForm();
+    if (!$contact_form) {
+      return [];
+    }
+
+    /** @var \Drupal\contact\MessageInterface $contact_message */
+    $message = $this->entityTypeManager->getStorage('contact_message')->create([
+      'contact_form' => $contact_form->id(),
+    ]);
+
+    $form = $this->entityFormBuilder->getForm($message, 'corporate_default');
+
+    return $form;
+  }
+
+  /**
+   * Returns the derived contact form.
+   *
+   * @return \Drupal\contact\Entity\ContactForm|null
+   *   The contact form entity.
+   */
+  protected function getContactForm(): ?ContactForm {
```

Comment:
Always type hint interfaces not classes.

Reply:
indeed thx